### PR TITLE
contract event ruby

### PR DESCRIPTION
**Pull request by jinghanx**
### What changed? Why?
contract event ruby

### CODE REVIEW

File: lib/coinbase/smart_contract.rb [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    # @param to_block_height [Integer] The end block height
+    # @return [Enumerable<Coinbase::ContractEvent>] The contract events
+    def self.list_events(
+      network_id,
+      protocol_name,
+      contract_address,
+      contract_name,
+      event_name,
+      from_block_height,
+      to_block_height
```
Comment (alex-stone) [inline lib/coinbase/smart_contract.rb:24]: With this many parameters we should make these keyword arguments

File: lib/coinbase/smart_contract.rb [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        event_name,
+        from_block_height,
+        to_block_height,
+        page
+      )
+    end
+  end
+
+  # Represents a single contract event
+  class ContractEvent
```
Comment (alex-stone) [inline lib/coinbase/smart_contract.rb:110]: Please put this in its own file

File: spec/unit/coinbase/smart_contract_spec.rb [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        { next_page: 'next_page' }
+      )
+    end
+
+    it 'returns an enumerator' do
+      expect(list_events).to be_an(Enumerator)
+    end
+  end
+
+  describe Coinbase::ContractEvent do
```
Comment (alex-stone) [inline spec/unit/coinbase/smart_contract_spec.rb:73]: This should live in `spec/unit/coinbase/contract_event_spec.rb`

**Review by alex-stone — APPROVED** [review]

**Comment by alex-stone** [comment]
Can we update the `Changelog.md` with this (can be in a follow-up PR if needed)

**Comment by jinghanx** [comment]
> Can we update the `Changelog.md` with this (can be in a follow-up PR if needed)

I do plan another PR to add some doc related stuff

**Pull request merged**